packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -350,11 +355,46 @@ export class RichTextEditor extends FoundationElement implements ErrorPattern {
                 Placeholder.configure({
                     placeholder: '',
                     showOnlyWhenEditable: false
+                }),
+                customLink.configure({
+                    // HTMLAttribute cannot be in camelCase as we want to match it with the name in Tiptap
+                    // eslint-disable-next-line @typescript-eslint/naming-convention
+                    HTMLAttributes: {
+                        rel: 'noopener noreferrer',
+                        target: null
+                    },
+                    openOnClick: false,
+                    linkOnPaste: false,
+                    validate: href => /^https?:\/\//.test(href)
                 })
             ]
         });
     }
 
+    /**
+     * Extending the default link mark schema defined in the TipTap.
+     *
+     * "exclude": https://prosemirror.net/docs/ref/#model.MarkSpec.excludes
+     * "inclusive": https://prosemirror.net/docs/ref/#model.MarkSpec.inclusive
+     * "parseHTML": https://tiptap.dev/guide/custom-extensions#parse-html
+     * "renderHTML": https://tiptap.dev/guide/custom-extensions/#render-html
+     */
+    private getCustomLinkExtension(): Mark<LinkOptions> {
+        return Link.extend({
+            excludes: '_',
```

Comment:
This `excludes: '_'` is an intentional configuration due to the markdown output we support. As the [Autolink](https://spec.commonmark.org/0.30/#autolinks) does not allow other formatting within the link, we have to restrict adding other formatting to the absolute link in the editor too. 

> Would there be a way to force `nimble-anchor` to be the outermost element if we have a bold/italic link?

Yes, this is how the default priority of the links will be established in relation to other formatting options such as bold and italics. However, this can only be enabled once we have implemented support for [Links(CommonMark doc)](https://spec.commonmark.org/0.30/#links) which are in a format like this: `[Lin text](https://foo)`. At present, we don't support this in markdown and will be planned in future updates as mentioned in the [spec](https://github.com/ni/nimble/tree/main/packages/nimble-components/src/rich-text/specs#:~:text=Support%20for%20adding%20hyperlinks%20to%20the%20existing%20text%20in%20the%20editor.%20This%20allows%20users%20to%20add%20links%20to%20existing%20text%20in%20the%20editor.%20When%20the%20link%20button%20in%20the%20formatting%20options%20is%20clicked%2C%20a%20dialog%20opens%2C%20providing%20a%20space%20to%20enter%20the%20hyperlink%20for%20the%20selected%20text.). Once it is enabled, we can remove this `excludes: '_'` from both the editor config and the parser schema. At that time, if we set the link to a bold text the markdown output will be something like `[**text**](https://foo)`